### PR TITLE
Supports the ability to honor `.gitignore` patterns (TOP LEVEL ONLY)

### PR DESCRIPTION
**Pull request by TerminalFi**
Attempt to have LSP-copilot honor `.gitignore` files that are detected at the top level of the window folders. Comparing the View path with the window path to get the correct ignore file.

### CODE REVIEW

**Comment by jfcherng** [comment]
> `.gitignore` files that are detected at the top level of the window folders.

Arguably this is not how `.gitignore` works.

**Comment by TerminalFi** [comment]
> > `.gitignore` files that are detected at the top level of the window folders.
> 
> Arguably this is not how `.gitignore` works. It should be searched bottom-up until `.git` is seen even if it's not in the workspace. (what if there is `.gitignore` but there is no `.git`? 😂)
> 
> Ok. I see **TOP LEVEL ONLY**.

Yes this is a very basic implementation, I have a more complete one in the works. But I also am expecting copilot to release a spec for this so not sure how much effort to put into this. 

At the moment I can push more which does searching bottom up until it reaches the workspace. Those changes will land soon.

**Comment by jfcherng** [comment]
> Yes this is a very basic implementation, I have a more complete one in the works. But I also am expecting copilot to release a spec for this so not sure how much effort to put into this.

Yes, making a feature request to Copilot is an option too.

**Comment by rchl** [comment]
Also it's not correct to treat all entries in .gitignore as paths relative to its directory. If the entry doesn't start with a slash then it applies also in subdirectories. and there are probably more rules, i haven't read the spec.

**Comment by TerminalFi** [comment]
> Also it's not correct to treat all entries in .gitignore as paths relative to its directory. If the entry doesn't start with a slash then it applies also in subdirectories. and there are probably more rules, i haven't read the spec.

Yup, I’ve got this solved as well I believe

File: LSP-copilot.sublime-settings [hunk 1]
```diff
@@ -16,7 +16,8 @@
 		"local_checks": false,
 		"telemetry": false,
 		"proxy": "",
-		"completion_style": "popup"
+		"completion_style": "popup",
+		"respect_gitignore": false
```
Comment (timfjord) [inline LSP-copilot.sublime-settings:20]: I would add a comment here and in the `sublime-package.json`'s `markdownDescription` section that the lookup is top level only

**Review by timfjord — APPROVED** [review]

File: LSP-copilot.sublime-settings (hunk 1 above)
Comment (TerminalFi) [reply LSP-copilot.sublime-settings:20]: I'll be pushing a change shortly that should have it functioning pretty closely to gitignore spec

**Comment by TerminalFi** [comment]
Copilot will natively support it soon.

**Pull request closed without merging**